ingest/anndata_.py

```diff
@@ -61,49 +55,76 @@ def generate_cluster_header(adata, clustering_name):
         else:
             msg = f"Too few dimensions for visualization in obsm \"{clustering_name}\", found {clustering_dimension}, expected 2 or 3."
             raise ValueError(msg)
-        with open(f"{clustering_name}.cluster.anndata_segment.tsv", "w") as f:
+        file_prefix = f"{study_file_id}.{clustering_name}"
+        with open(AnnDataIngestor.set_clustering_filename(file_prefix), "w") as f:
             f.write('\t'.join(headers) + '\n')
 
     @staticmethod
-    def generate_cluster_type_declaration(adata, clustering_name):
+    def generate_cluster_type_declaration(adata, clustering_name, study_file_id):
         """
         Based on clustering dimensions, write clustering TYPE line to file
         """
         clustering_dimension = adata.obsm[clustering_name].shape[1]
         types = ["TYPE", *["numeric"] * clustering_dimension]
-        with open(f"{clustering_name}.cluster.anndata_segment.tsv", "a") as f:
+        file_prefix = f"{study_file_id}.{clustering_name}"
+        with open(AnnDataIngestor.set_clustering_filename(file_prefix), "a") as f:
             f.write('\t'.join(types) + '\n')
 
     @staticmethod
-    def generate_cluster_body(adata, clustering_name):
+    def generate_cluster_body(adata, clustering_name, study_file_id):
         """
         Append clustering data to clustering file
         """
         cluster_cells = pd.DataFrame(adata.obs_names)
         cluster_body = pd.concat(
             [cluster_cells, pd.DataFrame(adata.obsm[clustering_name])], axis=1
         )
+        file_prefix = f"{study_file_id}.{clustering_name}"
         pd.DataFrame(cluster_body).to_csv(
-            AnnDataIngestor.set_output_filename(clustering_name),
+            AnnDataIngestor.set_clustering_filename(file_prefix),
             sep="\t",
             mode="a",
             header=None,
             index=False,
         )
 
     @staticmethod
-    def files_to_delocalize(arguments):
-        # ToDo - check if names using obsm_keys need sanitization
-        cluster_file_names = [AnnDataIngestor.set_output_filename(name) for name in arguments["obsm_keys"]]
-        return cluster_file_names
+    def set_clustering_filename(name):
+        return f"{name}.cluster.anndata_segment.tsv"
 
     @staticmethod
-    def set_output_filename(name):
-        return f"{name}.cluster.anndata_segment.tsv"
+    def generate_metadata_file(adata, output_name):
+        """
+        Generate metadata NAME and TYPE lines
+        """
+        headers = adata.obs.columns.tolist()
+        types = []
+        for header in headers:
+            if pd.api.types.is_number(adata.obs[header]):
+                types.append("NUMERIC")
+            else:
+                types.append("GROUP")
+        headers.insert(0, "NAME")
```

Comment:
Non-blocking: seems like this could be extracted out to a separate method and used for both clustering & metadata file generation.  We could call if from here and in places like `generate_cluster_header` or `generate_cluster_type_declaration`:
```
def write_output_file_header(output_name, header_type, headers)
    values = headers.insert(0, header_type)
    with open(output_name, "w") as f:
        f.write('\t'.join(values) + '\n')
```